src/lib/core/TLVReader.cpp

```diff
@@ -300,14 +300,27 @@ CHIP_ERROR TLVReader::Get(ByteSpan & v)
 
 CHIP_ERROR TLVReader::Get(CharSpan & v)
 {
+    constexpr int kAsciiUnitSeparator = 0x1F;
+
     if (!TLVTypeIsUTF8String(ElementType()))
     {
         return CHIP_ERROR_WRONG_TLV_TYPE;
     }
 
     const uint8_t * bytes;
     ReturnErrorOnFailure(GetDataPtr(bytes)); // Does length sanity checks
-    v = CharSpan(Uint8::to_const_char(bytes), GetLength());
+
+    uint32_t len = GetLength();
+
+    // If ASCII Unit Separator (0x1f) is present in the string then method returns string ending
```

Comment:
Let's call it Information Separator 1 to match Unicode and spec text

Reply:
done